crates/hyperqueue/src/client/output/cli.rs

```diff
@@ -534,6 +535,66 @@ impl Output for CliOutput {
         );
     }
 
+    fn print_job_cat(&self, job: JobDetail, stdout: bool, stderr: bool) {
+        let JobDetail {
+            info,
+            job_desc,
+            tasks,
+            max_fails: _,
+            submission_date,
+            completion_date_or_now,
+            submit_dir,
+        } = job;
+        let JobDescription::Array { ids, .. } = &job_desc;
+        let (resources, program_def, time_limit, priority, pin) = match &job_desc {
+            JobDescription::Array { task_desc, .. } => (
+                &task_desc.resources,
+                &task_desc.program,
+                task_desc.time_limit,
+                task_desc.priority,
+                task_desc.pin,
+            ),
+        };
```

Comment:
Just access stdout/stderr of the selected tasks, no need for this large amount of code :)
Don't use the stdout/stderr of the job, it might contain unresolved paths with placeholders.